### PR TITLE
Redux in components

### PR DESCRIPTION
**Pull request by levy002**
## Description
In this PR, we used the Redux store to display books.

### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### Milestone requirements, We:

- [x] connected the app to the Redux store.
- [x]  displayed books from the Redux store.
- [x] used `useSelector` to consume the states.
- [x] used `UseDispatch` to dispatch Actions.
- [x] kept states immutable in reducers.

 # Happy reviewing 😃

### CODE REVIEW

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @levy002, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/bookList/BookList.js [hunk 1]
```diff
@@ -1,49 +1,22 @@
...
-  ];
-
+  const books = useSelector((state) => state.BooksReducer);
   return (
     <>
       <ul className="allBooks">
-        <li className="bookInfo">
-          {books.map((book) => (
+        {
+          books.lenght === 0 ? books.map((book) => (
```
[the remark this answers is not in the document]
Comment (levy002) [reply src/components/bookList/BookList.js]: Thank you @Deepakdanger for mentioning this. It's solved.

File: src/components/form/Form.js [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
+      id: uuidv4(),
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook(book));
+    document.querySelector('form').reset();
+  };
```
[the remark this answers is not in the document]
Comment (levy002) [reply src/components/form/Form.js]: Thank you @Deepakdanger for mentioning this. It's solved.

**Review by bimbolabuari — APPROVED** [review]
Hi @levy002, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**